base/src/user_commands/signed_command/mod.rs

```diff
@@ -0,0 +1,270 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0
+
+//! Signed commands are commands that require signing with some accounts private key
+
+pub mod builder;
+
+use crate::numbers::{AccountNonce, Amount, GlobalSlotNumber, TokenId};
+use crate::user_commands::memo::SignedCommandMemo;
+use crate::user_commands::payment::PaymentPayload;
+
+use proof_systems::mina_hasher::{Hashable, ROInput};
+use proof_systems::mina_signer::{CompressedPubKey, Keypair, NetworkId, Signature, Signer};
+
+const TAG_BITS: usize = 3;
+const PAYMENT_TX_TAG: [bool; TAG_BITS] = [false, false, false];
+
+/// Top level signed command type
+#[derive(Clone, PartialEq, Debug)]
+pub struct SignedCommand {
+    /// The payload to sign
+    pub payload: SignedCommandPayload,
+    /// The signer (public key)
+    pub signer: CompressedPubKey,
+    /// The signature (result of signing payload with public key)
+    pub signature: Signature,
+}
+
+impl SignedCommand {
+    /// Sign a SignedCommandPayload to construct a SignedCommand
+    pub fn from_payload(
+        payload: SignedCommandPayload,
+        keypair: Keypair,
+        network: NetworkId,
+    ) -> Self {
+        // This should change to create_kimchi after fork
```

Comment:
For later: can probably use feature flags to identify forks and accordingly the code.